helix-term/src/application.rs

```diff
@@ -1134,6 +1134,63 @@ impl Application {
                         }
                         Ok(serde_json::Value::Null)
                     }
+                    Ok(MethodCall::ShowDocument(params)) => {
+                        let language_server = language_server!();
+                        let offset_encoding = language_server.offset_encoding();
+
+                        if params.external.unwrap_or_default() {
+                            self.jobs
+                                .callback(crate::open_external_url_callback(params.uri));
+                            return;
+                        }
+
+                        let path = match params.uri.to_file_path() {
+                            Ok(path) => path,
+                            Err(_) => {
+                                log::error!("Unsupported file URI: {}", params.uri);
+                                return;
+                            }
+                        };
+
+                        let action = match params.take_focus {
+                            Some(true) => helix_view::editor::Action::Replace,
+                            _ => helix_view::editor::Action::HorizontalSplit,
```

Comment:
hmm I am not sure a hsplit is always preferable that will depend on the exact screen layout and also personal preference.

I would expect a vsplit to be the more common choice (since it would allow side by side comparisons of files)?